src/rust/engine/fs/src/store.rs

```diff
@@ -370,27 +542,48 @@ mod local {
     }
   }
 
+  #[derive(Eq, PartialEq)]
+  struct AgedFingerprint {
+    pub fingerprint: Fingerprint,
+    pub size_bytes: usize,
+    pub database: Database,
+    pub expired_seconds_ago: u64,
+  }
+
+  impl PartialOrd for AgedFingerprint {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+      self.expired_seconds_ago.partial_cmp(
+        &other.expired_seconds_ago,
+      )
+    }
+  }
+
+  impl Ord for AgedFingerprint {
```

Comment:
Hmm, looks like maybe rust falls into the same trap as some other languages with `Ord`: PartialOrd, Ord and PartialEq need to be consistent: https://doc.rust-lang.org/std/cmp/trait.Ord.html#how-can-i-implement-ord

Reply:
Ugh. Moved it around a little to derive all three.